apis/v1alpha1/convert_test.go

```diff
@@ -48,9 +47,12 @@ service:
       processors: [resourcedetection]
       exporters: [otlp]
 `
+
+func Test_v1alpha1tov1beta1(t *testing.T) {
+	t.Run("valid config", func(t *testing.T) {
 		cfgV1 := OpenTelemetryCollector{
 			Spec: OpenTelemetryCollectorSpec{
```

Comment:
I'd love to see one full smoke test where we set every field in the spec.

Reply:
added